CSUB-387: Fix pallet discovery script

atodorov opened the pull request:
Description of proposed changes:


Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

atodorov committed:
Update wrong command in CI job

inside ci.yml we need to compare the list of pallets for which dummy
weights are being generated in the test job. This list comes from ci.yml
not from bench.yml!

atodorov committed:
Teach helper script to print discovered pallets and exit

this is the canonical list of pallets which we should compare against
everywhere else!

NOTE: only pallets which implement extrinsics are discovered by this
script. Pallets which don't expose extrinsics are ignored!

atodorov committed:
Use the canonical pallet source for comparisons in CI

atodorov committed:
Don't benchmark pallets which don't have extrinsics

atodorov committed:
Adjust pallet & extrinsic discovery script for staking pallet

b/c it introduces new locations and the search needs to be more flexible

CAGS295 commented on scripts/check-parity-bw-extrinsics-benchmarks-weights.sh:
```diff
@@ -1,9 +1,23 @@
 
 set -e
+
+FILES_WITH_EXTRINSICS=$(grep "pallet::call" pallets/* -R | cut -f1 -d: | sort | uniq)
```
Not only calls need benchmarks, hooks like `on_initialize` as well.

atodorov replied:
Will follow up on this in CSUB-389.

atodorov committed:
Rename CI config file

atodorov committed:
Make CI script more verbose

sometimes downloading latest release fails and I want to know what
exactly happened when there is a failure
